dev/src/codewind/connection/MCSocket.ts

```diff
@@ -248,6 +249,15 @@ export default class MCSocket implements vscode.Disposable {
         return project.onSettingsChangedEvent(payload);
     }
 
+    private readonly onLoadRunnerStatusChanged = async (payload: {projectID: string, status: string, timestamp: string}):
+        Promise<void> => {
+        const project = await this.getProject(payload);
+        if (project == null) {
+            return;
+        }
+        project.onLoadRunnerUpdate(payload);
```

Comment:
To make sure we don't get any unhandled promise rejections, you can either make sure every line with a potential error in `onLoadRunnerUpdate` is wrapped in a try-catch, or you can put a catch-all around this line. You'll have to `await project.onLoadRunnerUpdate` so that the `catch` block receives the promise rejection.

Reply:
Opted for a try/catch around `onLoadRunnerUpdate()` call. Have awaited.